src/test/java/org/alfresco/rest/api/tests/GroupsTest.java

```diff
@@ -1856,4 +1857,105 @@ private Group generateSubGroup(Group parentGroup)
 
         return subGroup;
     }
+
+    private void testGetGroupsWithDisplayNameFilter() throws Exception 

```

Comment:
Does the authorityService.getAuthoritiesInfo() method accept wildcards for display name? If so, we should have a test to ensure multiple results are returned.

Reply:
At the moment my code is not consisent, it  doesn't support wildcards in the case the parameter isRoot=true because in this specific case we don't use the canned query but the already loaded list of root authorities. 
I think it's better I change that and I avoid to use getAuthoritiesInfo to filter by displayName . It accepts wildcards (we don't need them) and if I want to support them I'll have to duplicate all the logic we have in [GetAuthoritiesCannedQuery.java](https://github.com/Alfresco/alfresco-repository/blob/5e247e50b48eeda597eefada2cc990b1797e3689/src/main/java/org/alfresco/repo/security/authority/GetAuthoritiesCannedQuery.java).
If use getAuthoritiesInfo() with displayNamefilter null as it was, I'll have to filter the result before the pagination. Is this a better solution or should I find a way to support wildcards when using the already loaded list of root authorities?

Reply:
For consistency we need to support display name filtering for root groups if we support it for non-root groups.

Looking at the JavaDoc for the getAuthoritiesInfo method (https://github.com/Alfresco/alfresco-repository/blob/ac38ac94ff4f9cbdf2671a9517781bda389a13c4/src/main/java/org/alfresco/service/cmr/security/AuthorityService.java#L169) it implies the name filter is only a startsWith case insensitive check (there are no wildcard characters passed) which should be fairly easy to replicate without having to duplicate the logic in GetAuthoritiesCannedQuery. 

In other words could you just modify the logic in displayNamePredicate here: https://github.com/Alfresco/alfresco-remote-api/blob/1f38ad40cf884a201755263ee93bcfcfa0872153/src/main/java/org/alfresco/rest/api/impl/GroupsImpl.java#L499 ?

Reply:
I tested it and it allows wildcards (eg. ?, .*), but the main problem is that it uses both the displayName and the authName to filter the result. I decided to not use it at the moment, the name filter will be a basic case insensitive comparation.

Reply:
OK